fluss-common/src/main/java/com/alibaba/fluss/row/arrow/ArrowWriter.java

```diff
@@ -138,7 +161,9 @@ public boolean isFull() {
             root.setRowCount(recordsCount);
             int metadataLength = getMetadataLength();
             int bodyLength = getBodyLength();
-            int currentSize = metadataLength + bodyLength;
+
+            int estimatedBodyLength = estimatedBytesWritten(bodyLength);
+            int currentSize = metadataLength + estimatedBodyLength;
```

Comment:
we also need to update `sizeInBytes()` to return the estimated size, otherwise, `RecordAccumulator#drain` can't respect the `maxSize` parameter.